source/asdmanager.py

```diff
@@ -48,25 +50,39 @@ def setup():
     ipaddresses = check_output("ip a | grep 'inet ' | sed 's/\s\s*/ /g' | cut -d ' ' -f 3 | cut -d '/' -f 1", shell=True).strip().splitlines()
     ipaddresses = [found_ip.strip() for found_ip in ipaddresses if found_ip.strip() != '127.0.0.1']
 
-    api_ip = Interactive.ask_choice(ipaddresses, 'Select the public IP address to be used for the API')
-    api_port = Interactive.ask_integer("Select the port to be used for the API", 1024, 65535, 8500)
-    ipaddresses.append('All')
-    asd_ips = []
-    add_ips = True
-    while add_ips:
-        current_ips = ' - Current selected IPs: {0}'.format(asd_ips)
-        new_asd_ip = Interactive.ask_choice(ipaddresses,
-                                            'Select an IP address or all IP addresses to be used for the ASDs{0}'.format(current_ips if len(asd_ips) > 0 else ''),
-                                            default_value='All')
-        if new_asd_ip == 'All':
-            ipaddresses.remove('All')
-            asd_ips = []
-            add_ips = False
-        else:
-            asd_ips.append(new_asd_ip)
-            ipaddresses.remove(new_asd_ip)
-            add_ips = Interactive.ask_yesno("Do you want to add another IP?")
-    asd_start_port = Interactive.ask_integer("Select the port to be used for the ASDs", 1024, 65535, 8600)
+    preconfig = '/tmp/openvstorage_preconfig.cfg'
+    if os.path.exists(preconfig):
+        config = RawConfigParser()
+        config.read(preconfig)
+        api_ip = config.get('asdmanager', 'api_ip')
+        api_port = int(config.get('asdmanager', 'api_port'))
+        asd_ips = config.get('asdmanager', 'asd_ips')
+        asd_start_port = int(config.get('asdmanager', 'asd_start_port'))
+
+        assert 1024 <= api_port <= 65535, "api_port should be >= 1024 and <= 65535"
+        assert 1024 <= asd_start_port <= 65535, "asd_start_port should be >= 1024 and <= 65535"
+        assert api_port not in range(asd_start_port, asd_start_port + 100), "api_port should not be in asd_port range:"\
+                                                                            "{0} - {} + 100".format(asd_start_port)
```

Comment:
This raises error:

In [3]: "api_port should not be in asd_port range: {0} - {} + 100".format(asd_start_port)
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-3-0b8fc5de3d53> in <module>()
----> 1 "api_port should not be in asd_port range: {0} - {} + 100".format(asd_start_port)

ValueError: cannot switch from manual field specification to automatic field numbering